templates/flavors/k3s/full-vpcless/kustomization.yaml

```diff
@@ -128,3 +128,35 @@ patches:
         labels:
           cluster-autoscaler: "true"
           etcd-backup: "true"
+  - target:
+      group: controlplane.cluster.x-k8s.io
+      version: v1beta1
+      kind: KThreesControlPlane
+    patch: |-
+      - op: replace
+        path: /spec/kthreesConfigSpec/preK3sCommands
+        value:
+          - |
+            mkdir -p /etc/rancher/k3s/config.yaml.d/
+            echo -n "kubelet-arg: \"--node-ip=" >> /etc/rancher/k3s/config.yaml.d/capi-config.yaml
+            echo -n "$(ip a s eth0 |grep -E 'inet '  |cut -d' ' -f6|cut -d/ -f1 | grep -E '192.168')" >> /etc/rancher/k3s/config.yaml.d/capi-config.yaml
+            echo ",$(ip a s eth0 |grep -E 'inet6 '  |cut -d' ' -f6|cut -d/ -f1 | grep -vE 'fe80')\"" >> /etc/rancher/k3s/config.yaml.d/capi-config.yaml
+          - sed -i '/swap/d' /etc/fstab
+          - swapoff -a
+          - hostnamectl set-hostname '{{ ds.meta_data.label }}' && hostname -F /etc/hostname
+  - target:
+      group: bootstrap.cluster.x-k8s.io
+      version: v1beta1
+      kind: KThreesConfigTemplate
+    patch: |-
+      - op: replace
+        path: /spec/template/spec/preK3sCommands
+        value:
+          - |
+            mkdir -p /etc/rancher/k3s/config.yaml.d/
+            echo -n "kubelet-arg: \"--node-ip=" >> /etc/rancher/k3s/config.yaml.d/capi-config.yaml
+            echo -n "$(ip a s eth0 |grep -E 'inet '  |cut -d' ' -f6|cut -d/ -f1 | grep -E '192.168')" >> /etc/rancher/k3s/config.yaml.d/capi-config.yaml
+            echo ",$(ip a s eth0 |grep -E 'inet6 '  |cut -d' ' -f6|cut -d/ -f1 | grep -vE 'fe80')\"" >> /etc/rancher/k3s/config.yaml.d/capi-config.yaml
+          - sed -i '/swap/d' /etc/fstab
+          - swapoff -a
+          - hostnamectl set-hostname '{{ ds.meta_data.label }}' && hostname -F /etc/hostname
```

Comment:
We also need the nodeSelector for linode-ccm. https://github.com/linode/cluster-api-provider-linode/blob/main/templates/flavors/k3s/default/kustomization.yaml#L54-L55